GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1534 @@
+            // **************************************************************************************************
+            if (newApp.ObjectId != null)
+            {
+                ExtensionProperty linkedInUserId = new ExtensionProperty
+                {
+                    Name = extName,
+                    DataType = "String",
+                    TargetObjects = {"User"}
+                };
+                try
+                {
+                    // firstly, let's write out all the existing cloud extension properties in the tenant
+                    IEnumerable<IExtensionProperty> allExts = await client.GetAvailableExtensionPropertiesAsync(false);
+                    foreach (ExtensionProperty ext in allExts)
+                    {
+                        Console.WriteLine("\nExtension: {0}", ext.Name);
+                    }
+                    newApp.ExtensionProperties.Add(linkedInUserId);
+                    await newApp.UpdateAsync();
+                    Console.WriteLine("\nUser object extended successfully with extension: {0}.", linkedInUserId.Name);
+                }
+                catch (Exception e)
+                {
+                    Program.WriteError("\nError extending the user object {0} {1}", e.Message,
+                        e.InnerException != null ? e.InnerException.Message : "");
+                }
+                return linkedInUserId;
+
+                #endregion
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        private static async Task ManipulateExtensionProperty(Application newApp, string extName, User newUser)
+        {
+            #region Manipulate an Extension Property
+
+            // **************************************************************************************************
+            // Update an extension property that exists on the user entity
+            // **************************************************************************************************
+
+            // create the extension attribute name, for the extension that we just created
+            string attributeName = "extension_" + newApp.AppId + "_" + extName;
+            try
+            {
+                if (newUser != null && newUser.ObjectId != null)
+                {
+                    newUser.SetExtendedProperty(attributeName, "user@linkedin.com");
+                    await newUser.UpdateAsync();
+                    Console.WriteLine("\nUser {0}'s extended property set successully.", newUser.DisplayName);
+                }
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError Updating the user object {0} {1}", e.Message,
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+
+            #endregion
+        }
+
+        private static void PrintExtensionProperty(User user, string attributeName)
```

Comment:
Added check

---
In reply to: [80746927](https://github.com/Azure-Samples/active-directory-dotnet-graphapi-console/pull/37#discussion_r80746927) [](ancestors = 80746927)